src/cowrie/commands/ls.py

```diff
@@ -121,16 +124,34 @@ def do_ls_l(self, path: str) -> None:
         Display detailed information about files
         Mimics the output of GNU ls -l and supports the following options:
         -a, -h
-        
+
         :param path: The path to list
         """
         files = self.get_dir_files(path)
         if not files:
             return
-
+        # Create a list to hold formatted sizes for display
+        formatted_sizes = []
         filesize_str_extent = 0
         if len(files):
             filesize_str_extent = max(len(str(x[fs.A_SIZE])) for x in files)
+            if self.showHumanReadable:
+                for x in files:
+                    size = x[fs.A_SIZE]  # Original size
+                    if size >= 1024 * 1024 * 1024:
+                        formatted_size = f"{size / 1024 / 1024 / 1024:.1f}G"
+                    elif size >= 1024 * 1024:
+                        formatted_size = f"{size / 1024 / 1024:.1f}M"
+                    elif size >= 1024:
+                        formatted_size = f"{size / 1024:.1f}K"
+                    else:
+                        formatted_size = str(size)
+
+                    formatted_sizes.append(formatted_size)
+                # Recalculate the extent with the formatted sizes
+                filesize_str_extent = max(len(size) for size in formatted_sizes)
+            else:
+                formatted_sizes = [str(x[fs.A_SIZE]) for x in files]
```

Comment:
This ensures the usage of original file size values while using -l argument again.